### PR TITLE
Improve PowerShell Remoting Argument Validation

### PR DESCRIPTION
**Pull request by anamnavi**
# PR Summary
This pull request refactors SSH process argument handling in PowerShell remoting to use ProcessStartInfo.ArgumentList throughout the SSH invocation flow and expands test coverage for invalid and edge-case parameter values.

By treating SSH arguments as discrete entries instead of constructing command-line strings, the implementation becomes simpler, more consistent, and easier to maintain. The update also includes additional validation tests, platform-specific test improvements, and helper logic for Windows process creation.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
  - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge. If this PR is a work in progress, please open this as a [Draft Pull Request and mark it as Ready to Review when it is ready to merge](https://docs.github.com/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#draft-pull-requests).
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
  - [ ] None
  - **OR**
  - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.5/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
    - [ ] Experimental feature name(s): 
- **User-facing changes**
  - [ ] Not Applicable
  - **OR**
  - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [ ] Issue filed: 
- **Testing - New and feature**
  - [ ] N/A or can only be tested interactively
  - **OR**
  - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

File: src/System.Management.Automation/engine/remoting/common/RunspaceConnectionInfo.cs [hunk 1]
```diff
@@ -2841,6 +2809,90 @@ private static Process CreateProcessWithRedirectedStd(
...
+                    builder.Append(' ');
+                }
+
+                if (argument == null
+                    || argument.Contains(' ')
+                    || argument.Contains('"')
+                    || argument.Contains('\t')
+                    || argument.Contains('\n')
+                    || argument.Contains('\r'))
+                {
```
[the remark this answers is not in the document]
Comment (tgauth) [reply src/System.Management.Automation/engine/remoting/common/RunspaceConnectionInfo.cs:2833]: This is unnecessary because arguments will not be empty due to prior checks in the code

**Review by TravisEz13 — APPROVED** [review]
